### PR TITLE
[DOC] Add derivative work credits

### PR DESCRIPTION
**Pull request by HugoDelatte**
#### What does this implement/fix? Explain your changes.
Credits in the readme and document index were already given but I forgot to add them in their respective code files with proper copyright. They include reference of derivative works from scikit-learn, scikit-portfolio, Riskfolio-Lib and statsmodel (all under MIT or BSD 3)

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

Is the syntax correct for this kind of credits?

#### Did you add any tests for the change?

N.A.

#### Any other comments?
No

##### For all contributions
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.
- [ ] I've added myself to the [list of contributors](https://github.com/skfolio/skfolio/blob/main/CONTRIBUTORS.md)
  How to: add yourself to the [all-contributors file](https://github.com/skfolio/skfolio/blob/main/.all-contributorsrc) in the `skfolio` root directory (not the `CONTRIBUTORS.md`). 
- [ ] Optionally, I've added myself and possibly others to the [CODEOWNERS](https://github.com/skfolio/skfolio/blob/main/CODEOWNERS) file - do this if you want to become the owner or maintainer of an estimator you added.

##### For new estimators
- [ ] I've added the estimator to the API reference in `docs/api.rst`.
- [ ] I've added one or more illustrative usage examples to the docstring and the `examples` section.

### CODE REVIEW

File: src/skfolio/prior/_black_litterman.py
Comment (CarloNicolini) [inline src/skfolio/prior/_black_litterman.py:1]: Maybe here some reference to pyportfoliopt too?

File: src/skfolio/optimization/_base.py
Comment (CarloNicolini) [inline src/skfolio/optimization/_base.py:1]: probably here some reference from scikit-portfolio too with the fit and predict logic?

**Review by CarloNicolini — APPROVED** [review]
I believe all the changes make sense. Just added some observations.

File: src/skfolio/prior/_black_litterman.py
Comment (HugoDelatte) [reply src/skfolio/prior/_black_litterman.py:1]: agreed and added

File: src/skfolio/optimization/_base.py
Comment (HugoDelatte) [reply src/skfolio/optimization/_base.py:1]: agreed and added

**Comment by HugoDelatte** [comment]
Thx, remaining credits have been added

**Pull request merged**